### PR TITLE
CLE-NST-AJC tests for Currency

### PR DESCRIPTION
**Pull request by mikhailZakharevich**

### CODE REVIEW

File: src/test/scala/group/scala/karazin/circe/literal/extras/PlainPrimitiveEncodeSuite.scala [hunk 1]
```diff
@@ -1297,3 +1298,45 @@ class PlainPrimitiveEncodeSuite extends munit.ScalaCheckSuite:
...
     }
   }
+
+  property("inlined Currency value") {
+
+    case class Primitive(value: Currency) derives Codec.AsObject
+
+    extension (inline sc: StringContext)
+      inline def encode(inline args: Any*): Json =
+        $ {macros.encode[Currency]('sc, 'args)}
```
Comment (IgorWolkov) [inline src/test/scala/group/scala/karazin/circe/literal/extras/PlainPrimitiveEncodeSuite.scala]: Should be `Primitive` not `Currency`.

File: src/test/scala/group/scala/karazin/circe/literal/extras/PlainPrimitiveEncodeSuite.scala [hunk 2]
```diff
@@ -1297,3 +1298,45 @@ class PlainPrimitiveEncodeSuite extends munit.ScalaCheckSuite:
...
+
+    forAll { (currency: Currency) =>
+
+      val primitive = Primitive(currency)
+
+      val result: Json = currency.asJson
+
+      val expected: Json = encode"${primitive.value}"
```
Comment (IgorWolkov) [inline src/test/scala/group/scala/karazin/circe/literal/extras/PlainPrimitiveEncodeSuite.scala]: Due to the issue above, should be `{"value": ${primitive.value}}`

**Review by IgorWolkov — CHANGES_REQUESTED** [review]
Please fix the test's semantic.

File: src/test/scala/group/scala/karazin/circe/literal/extras/PlainPrimitiveEncodeSuite.scala (hunk 2 above)
Comment (mikhailZakharevich) [reply src/test/scala/group/scala/karazin/circe/literal/extras/PlainPrimitiveEncodeSuite.scala]: Done

File: src/test/scala/group/scala/karazin/circe/literal/extras/PlainPrimitiveEncodeSuite.scala (hunk 1 above)
Comment (mikhailZakharevich) [reply src/test/scala/group/scala/karazin/circe/literal/extras/PlainPrimitiveEncodeSuite.scala]: Done

**Review by IgorWolkov — APPROVED** [review]

**Pull request merged**